pkg/kv/kvserver/client_mvcc_gc_test.go

```diff
@@ -68,3 +89,198 @@ func TestMVCCGCCorrectStats(t *testing.T) {
+			overrideCfg := config
+			overrideCfg.GCPolicy.TTLSeconds = 1
+			return overrideCfg
+		},
+	}
+	tc := testcluster.StartTestCluster(t, 1, args)
+	defer tc.Stopper().Stop(ctx)
+	sqlDB := tc.ServerConn(0)
+
+	st := tc.Server(0).ClusterSettings()
+	gcjob.EmptySpanPollInterval.Override(ctx, &st.SV, 100*time.Millisecond)
+
+	// Speed up propagation of span configs and protected timestamps.
+	_, err := sqlDB.Exec("SET CLUSTER SETTING kv.closed_timestamp.target_duration = '100ms'")
+	require.NoError(t, err)
+	_, err = sqlDB.Exec("SET CLUSTER SETTING kv.protectedts.poll_interval = '10ms';")
+	require.NoError(t, err)
+
+	// Infrastructure to create a protected timestamp record.
+	execCfg := tc.Server(0).ExecutorConfig().(sql.ExecutorConfig)
+	ptp := execCfg.ProtectedTimestampProvider
+	insqlDB := execCfg.InternalDB
+	mkRecordAndProtect := func(ts hlc.Timestamp, target *ptpb.Target) *ptpb.Record {
+		recordID := uuid.MakeV4()
+		rec := jobsprotectedts.MakeRecord(recordID, int64(1), ts, nil, /* deprecatedSpans */
+			jobsprotectedts.Jobs, target)
+		require.NoError(t, insqlDB.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
+			return ptp.WithTxn(txn).Protect(ctx, rec)
+		}))
+		return rec
+	}
+
+	runner := sqlutils.MakeSQLRunner(sqlDB)
+	runner.Exec(t, `CREATE TABLE foo (id INT PRIMARY KEY)`)
+
+	var tableID int
+	runner.QueryRow(t, "SELECT table_id FROM crdb_internal.tables WHERE name = 'foo'").Scan(&tableID)
+	require.NotEqual(t, 0, tableID)
+	tablePrefix := keys.SystemSQLCodec.TablePrefix(uint32(tableID))
+	tc.SplitRangeOrFatal(t, tablePrefix)
+	require.NoError(t, tc.WaitForSplitAndInitialization(tablePrefix))
+
+	// Lower the GC TTL of table foo and allow the zone config to get reconciled.
+	s := tc.Server(0)
+	store, err := s.GetStores().(*kvserver.Stores).GetStore(s.GetFirstStoreID())
+	require.NoError(t, err)
+	repl := store.LookupReplica(roachpb.RKey(tablePrefix))
+	_, err = sqlDB.Exec(`ALTER TABLE foo CONFIGURE ZONE USING gc.ttlseconds = 1`)
+	require.NoError(t, err)
+	testutils.SucceedsSoon(t, func() error {
+		if repl.SpanConfig().GCPolicy.TTLSeconds != 1 {
+			return errors.New("waiting for span config to apply")
+		}
+		return nil
+	})
+
+	for i := 0; i < 10; i++ {
+		runner.Exec(t, `INSERT INTO foo VALUES ($1)`, i)
+	}
+
+	// Drop the table and wait for the MVCC GC schema change job to write its
+	// range deletion tombstone and start polling to check if the target span
+	// has been gc'ed.
+	runner.Exec(t, `DROP TABLE foo CASCADE `)
+	var jobID int64
+	runner.QueryRow(t, "SELECT job_id FROM [SHOW JOBS] WHERE job_type = 'SCHEMA CHANGE GC'").Scan(&jobID)
+	require.NotEqual(t, 0, jobID)
+	runner.CheckQueryResultsRetry(t, `
+SELECT count(*)
+  FROM (
+        SELECT job_id
+          FROM [SHOW JOBS]
+         WHERE job_type = 'SCHEMA CHANGE GC'
+               AND status = 'running'
+               AND running_status = 'waiting for MVCC GC'
+       )`,
+		[][]string{{"1"}})
+
```

Comment:
[nit] I guess one of the issues with the small intervals of this test is that technically at this point the schema GC Job might be done and a round of mvccGC could have happened, so the PTS record below would be erroneous.  

It's probably fine for now since I imagine the GC scoring will make that unlikely; although the gc drop hints that were just added may change that.

Perhaps just to avoid confusion, we could re-read the GC job state after adding the PTS record below just to be sure the test is still doing what we think it should.